src/ts/components/ads/AdCounterLabel.ts

```diff
@@ -0,0 +1,39 @@
+import { UIInstanceManager } from '../../UIManager';
+import {LabelConfig, Label} from '../labels/Label';
+import { PlayerAPI } from 'bitmovin-player';
+
+/**
+ * A label that displays the index of the currently playing ad out of the total number of ads.
+ *
+ * @category Labels
+ */
+export class AdCounterLabel extends Label<LabelConfig> {
+  constructor(config: LabelConfig = {}) {
+    super(config);
+
+    this.config = this.mergeConfig(config, {
+      cssClass: 'ui-label-ad-counter',
+    }, this.config);
+  }
+
+  configure(player: PlayerAPI, uimanager: UIInstanceManager): void {
+    super.configure(player, uimanager);
+
+    let currentAdIndex: number = 0;
+    let totalAdsCount: number = 0;
+
+    const reset = () => {
+      currentAdIndex = 0;
+      totalAdsCount = 0;
+      this.setText('');
+    };
+
+    player.on(player.exports.PlayerEvent.AdStarted, () => {
+      currentAdIndex++;
+      totalAdsCount = player.ads.getActiveAdBreak().ads?.length ?? 0;
+      this.setText(`Ad ${currentAdIndex} of ${totalAdsCount}`);
```

Comment:
Same as on the other PR: less state handling is better. In this case here, when we miss a reset in some events the `currentAdIndex` will just continue to grow. 

What about something like this: 🤔 
```suggestion
      const activeAdIndex = player.ads.getActiveAdBreak().ads.findIndex((ad) => ad === player.ads.getActiveAd()) + 1;
      const totalAdsCount = player.ads.getActiveAdBreak().ads?.length ?? 0;
      this.setText(`Ad ${activeAdIndex} of ${totalAdsCount}`);
```

Reply:
Good idea, and thanks for the snippet 💪 

Address in 1abb1f672e065823193b34b43d6e755d74b1ce0b together with the state variables removal